config/src/config/network_config.rs

```diff
@@ -121,14 +108,11 @@ impl NetworkConfig {
             self.listen_address = utils::get_local_ip().ok_or_else(|| anyhow!("No local IP"))?;
         }
 
-        let keypair = NetworkKeyPairs::default();
-        let default_peer_id = PeerId::try_from(keypair.identity_keys.public().to_bytes()).unwrap();
-        let identity_key = self.network_keypairs.identity_keys.public();
-        let key_peer_id = PeerId::try_from(identity_key.to_bytes()).unwrap();
-
-        // If PeerId is not set, it is derived from NetworkIdentityKey.
-        if self.peer_id == default_peer_id {
-            self.peer_id = key_peer_id;
+        if self.enable_remote_authentication {
+            ensure!(
+                self.enable_noise,
```

Comment:
probably also want to check if enable_noise network_keypairs is some

Reply:
Done, thanks!

Reply:
Hmm.. this check won't always hold (e.g. during testing, when defaults don't provide a value for the keys). I agree that we should ideally check the is_some() property here, but this will require a bigger change. Deferring this work to a future PR.